src/Features/VisualBasic/Portable/CodeRefactorings/InlineMethod/VisualBasicInlineMethodRefactoringProvider.vb

```diff
@@ -0,0 +1,70 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Imports System.Composition
+Imports Microsoft.CodeAnalysis.CodeRefactorings
+Imports Microsoft.CodeAnalysis.Editing
+Imports Microsoft.CodeAnalysis.Host.Mef
+Imports Microsoft.CodeAnalysis.InlineMethod
+Imports Microsoft.CodeAnalysis.VisualBasic.LanguageServices
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.CodeRefactorings.InlineTemporary
+
+    <ExportCodeRefactoringProvider(LanguageNames.VisualBasic, Name:=NameOf(PredefinedCodeRefactoringProviderNames.InlineMethod)), [Shared]>
+    <Export(GetType(VisualBasicInlineMethodRefactoringProvider))>
+    Friend Class VisualBasicInlineMethodRefactoringProvider
+        Inherits AbstractInlineMethodRefactoringProvider(Of InvocationExpressionSyntax, ExpressionSyntax, MethodBlockSyntax, StatementSyntax)
+
+        <ImportingConstructor>
+        <Obsolete(MefConstruction.ImportingConstructorMessage, True)>
+        Public Sub New()
+            MyBase.New(VisualBasicSyntaxFacts.Instance, VisualBasicSemanticFactsService.Instance)
+        End Sub
+
+        Protected Overrides Function GetInlineExpression(methodBlock As MethodBlockSyntax) As ExpressionSyntax
+            Dim statements = methodBlock.Statements
+            If statements.Count = 1 Then
+                Dim singleStatement = statements(0)
+                Dim returnStatement = TryCast(singleStatement, ReturnStatementSyntax)
+                If returnStatement IsNot Nothing Then
+                    Return returnStatement.Expression
+                End If
+
+                Dim expressionStatement = TryCast(singleStatement, ExpressionStatementSyntax)
+                If expressionStatement IsNot Nothing Then
+                    Return expressionStatement.Expression
+                End If
+            End If
+            Return Nothing
```

Comment:
i want to say that all of this should be possible to do in the base class, just using ISyntaxFacts.

Reply:
Ah, there is a slight difference between C# and VB. Like C# support arrow function & throw expression so I put them in different parts. So is that no enough difference?

Reply:
wfm.